### PR TITLE
Better exercise var name in tests

### PR DESCRIPTION
**Pull request by Bajger**
Changes related to discussion topic: https://forum.exercism.org/t/variable-naming/10172/6 
- Ongoing changes will redefine variable name on all active exercises. 
- Active exercises will get updated metadata and descriptions as well.

Current status is that many exercise descriptions are outdated/out-of-sync.
NOTE: **Changes in exercises directory will be in separate PR** (to not mix generator changes and exercise changes).  
```
./bin/configlet sync
Updating cached 'problem-specifications' data...
Checking exercises...
[warn] docs: instructions unsynced: darts
[warn] docs: instructions unsynced: flatten-array
[warn] docs: instructions unsynced: matching-brackets
[warn] docs: introduction missing: matching-brackets
[warn] docs: instructions unsynced: minesweeper
[warn] docs: introduction missing: minesweeper
[warn] docs: instructions unsynced: space-age
[warn] docs: introduction missing: space-age
[warn] metadata: unsynced: darts
[warn] metadata: unsynced: raindrops
[warn] allergies: missing 1 test case
       - list when: -> no allergen score parts without highest valid score (93c2df3e-4f55-4fed-8116-7513092819cd)
[warn] anagram: missing 8 test cases
       - detects two anagrams (03eb9bbe-8906-4ea0-84fa-ffe711b52c8b)
       - does not detect an anagram if the original word is repeated (630abb71-a94e-4715-8395-179ec1df9f91)
       - words are not anagrams of themselves (68934ed0-010b-4ef9-857a-20c9012d1ebf)
       - words are not anagrams of themselves even if letter case is partially different (589384f3-4c8a-4e7d-9edc-51c3e5f0c90e)
       - words are not anagrams of themselves even if letter case is completely different (ba53e423-7e02-41ee-9ae2-71f91e6d18e6)
       - words other than themselves can be anagrams (33d3f67e-fbb9-49d3-a90e-0beb00861da7)
       - handles case of greek letters (a6854f66-eec1-4afd-a137-62ef2870c051)
       - different characters may have the same bytes (fd3509e5-e3ba-409d-ac3d-a9ac84d13296)
[warn] armstrong-numbers: missing 2 test cases
       - Armstrong number containing seven zeroes (5ee2fdf8-334e-4a46-bb8d-e5c19c02c148)
       - The largest and last Armstrong number (12ffbf10-307a-434e-b4ad-c925680e1dd4)
[warn] bowling: missing 1 test case
       - last two strikes followed by only last bonus with non strike points (efb426ec-7e15-42e6-9b96-b4fca3ec2359)
[warn] collatz-conjecture: missing 2 test cases
       - zero is an error (2187673d-77d6-4543-975e-66df6c50e2da)
       - negative value is an error (ec11f479-56bc-47fd-a434-bcd7a31a7a2e)
[warn] flatten-array: missing 5 test cases
       - empty (8c71dabd-da60-422d-a290-4a571471fb14)
       - flattens a nested array (3f15bede-c856-479e-bb71-1684b20c6a30)
       - null values are omitted from the final result (0705a8e5-dc86-4cec-8909-150c5e54fa9c)
       - consecutive null values at the front of the list are omitted from the final result (c6cf26de-8ccd-4410-84bd-b9efd88fd2bc)
       - consecutive null values in the middle of the list are omitted from the final result (382c5242-587e-4577-b8ce-a5fb51e385a1)
[warn] forth: missing 3 test cases
       - parsing and numbers -> pushes negative numbers onto the stack (fd7a8da2-6818-4203-a866-fed0714e7aa0)
       - user-defined words -> cannot redefine negative numbers (df5b2815-3843-4f55-b16c-c3ed507292a7)
       - user-defined words -> only defines locally (3c8bfef3-edbb-49c1-9993-21d4030043cb)
[warn] grade-school: missing 16 test cases
       - Add a student (9337267f-7793-4b90-9b4a-8e3978408824)
       - Adding multiple students in the same grade in the roster (73c3ca75-0c16-40d7-82f5-ed8fe17a8e4a)
       - Cannot add student to same grade in the roster more than once (87c871c1-6bde-4413-9c44-73d59a259d83)
       - A student can't be in two different grades (c125dab7-2a53-492f-a99a-56ad511940d8)
       - A student can only be added to the same grade in the roster once (a0c7b9b8-0e89-47f8-8b4a-c50f885e79d1)
       - Student not added to same grade in the roster more than once (d7982c4f-1602-49f6-a651-620f2614243a)
       - Adding students in multiple grades (e70d5d8f-43a9-41fd-94a4-1ea0fa338056)
       - Cannot add same student to multiple grades in the roster (7df542f1-57ce-433c-b249-ff77028ec479)
       - A student cannot be added to more than one grade in the sorted roster (6a03b61e-1211-4783-a3cc-fc7f773fba3f)
       - Student not added to multiple grades in the roster (c7ec1c5e-9ab7-4d3b-be5c-29f2f7a237c5)
       - Students are sorted by grades in the roster (d9af4f19-1ba1-48e7-94d0-dabda4e5aba6)
       - Students are sorted by name in the roster (d9fb5bea-f5aa-4524-9d61-c158d8906807)
       - Grade is empty if no students in that grade (1e0cf06b-26e0-4526-af2d-a2e2df6a51d6)
       - Student not added to same grade more than once (2bfc697c-adf2-4b65-8d0f-c46e085f796e)
       - Student not added to multiple grades (66c8e141-68ab-4a04-a15a-c28bc07fe6b9)
       - Student not added to other grade for multiple grades (c9c1fc2f-42e0-4d2c-b361-99271f03eda7)
[warn] hamming: missing 6 test cases
       - disallow first strand longer (b9228bb1-465f-4141-b40f-1f99812de5a8)
       - disallow second strand longer (dab38838-26bb-4fff-acbe-3b0a9bfeba2d)
       - disallow left empty strand (db92e77e-7c72-499d-8fe6-9354d2bfd504)
       - disallow empty first strand (b764d47c-83ff-4de2-ab10-6cfe4b15c0f3)
       - disallow right empty strand (920cd6e3-18f4-4143-b6b8-74270bb8f8a3)
       - disallow empty second strand (9ab9262f-3521-4191-81f5-0ed184a5aa89)
[warn] high-scores: missing 4 test cases
       - Top 3 scores -> Latest score after personal top scores (2df075f9-fec9-4756-8f40-98c52a11504f)
       - Top 3 scores -> Scores after personal top scores (809c4058-7eb1-4206-b01e-79238b9b71bc)
       - Top 3 scores -> Latest score after personal best (ddb0efc0-9a86-4f82-bc30-21ae0bdc6418)
       - Top 3 scores -> Scores after personal best (6a0fd2d1-4cc4-46b9-a5bb-2fb667ca2364)
[warn] isbn-verifier: missing 2 test cases
       - invalid check digit in isbn is not treated as zero (9416f4a5-fe01-4b61-a07b-eb75892ef562)
       - invalid characters are not ignored before checking length (daad3e58-ce00-4395-8a8e-e3eded1cdc86)
[warn] isogram: missing 1 test case
       - word with duplicated character and with two hyphens (0d0b8644-0a1e-4a31-a432-2b3ee270d847)
[warn] luhn: missing 4 test cases
       - invalid long number with a remainder divisible by 5 (7e7c9fc1-d994-457c-811e-d390d52fba5e)
       - very long input is valid (b9887ee8-8337-46c5-bc45-3bcab51bc36f)
       - valid luhn with an odd number of digits and non zero first digit (8a7c0e24-85ea-4154-9cf1-c2db90eabc08)
       - non-numeric, non-space char in the middle with a sum that's divisible by 10 isn't allowed (8b72ad26-c8be-49a2-b99c-bcc3bf631b33)
[warn] matching-brackets: missing 3 test cases
       - paired and wrong nested brackets but innermost are correct (1d5c093f-fc84-41fb-8c2a-e052f9581602)
       - early unexpected brackets (a345a753-d889-4b7e-99ae-34ac85910d1a)
       - early mismatched brackets (21f81d61-1608-465a-b850-baa44c5def83)
[warn] pangram: missing 1 test case
       - a-m and A-M are 26 different characters but not a pangram (7138e389-83e4-4c6e-8413-1e40a0076951)
[warn] reverse-string: missing 3 test cases
       - wide characters (1bed0f8a-13b0-4bd3-9d59-3d0593326fa2)
       - grapheme cluster with pre-combined form (93d7e1b8-f60f-4f3c-9559-4056e10d2ead)
       - grapheme clusters (1028b2c1-6763-4459-8540-2da47ca512d9)
[warn] roman-numerals: missing 8 test cases
       - 16 is XVI (6d1d82d5-bf3e-48af-9139-87d7165ed509)
       - 66 is LXVI (4465ffd5-34dc-44f3-ada5-56f5007b6dad)
       - 166 is CLXVI (902ad132-0b4d-40e3-8597-ba5ed611dd8d)
       - 666 is DCLXVI (dacb84b9-ea1c-4a61-acbb-ce6b36674906)
       - 1666 is MDCLXVI (efbe1d6a-9f98-4eb5-82bc-72753e3ac328)
       - 3001 is MMMI (3bc4b41c-c2e6-49d9-9142-420691504336)
       - 3888 is MMMDCCCLXXXVIII (2f89cad7-73f6-4d1b-857b-0ef531f68b7e)
       - 3999 is MMMCMXCIX (4e18e96b-5fbb-43df-a91b-9cb511fe0856)
[warn] space-age: missing 1 test case
       - invalid planet causes error (57b96e2a-1178-40b7-b34d-f3c9c34e4bf4)
[warn] tournament: missing 1 test case
       - ensure points sorted numerically (f9e20931-8a65-442a-81f6-503c0205b17a)
[warn] word-count: missing 2 test cases
       - with apostrophes (4ff6c7d7-fcfc-43ef-b8e7-34ff1837a2d3)
       - quotation for word with apostrophe (6d00f1db-901c-4bec-9829-d20eb3044557)
[warn] some exercises have unsynced docs
[warn] some exercises have unsynced metadata
[warn] some exercises are missing test cases
```

### CODE REVIEW

**Review by glennj — APPROVED** [review]

**Comment by Bajger** [comment]
@glennj : PR is ready to review (if you will). Sorry for this extensive list of commits. I just needed to dig in more and more to fix to work with v3. Any thoughts, comments, let me know. Otherwise it should be pretty unit tested.
TLDR: All generator classes have been updated, so it doesn't really affect anything in exercise directory. These changes will allow to update exercises and regenerate metadata smoothly. 

What will follow?
I will do on unsynced exercises (listed above) an update: 
1. sync from problem specifications to Pharo (update tests) by e.g.:
`ExercismExerciseGenerator new regenerateExerciseFrom: '/Users/dbajger/Documents/Pharo/problem-specifications/exercises/darts' asFileReference`
2. Export updated exercise sources to exercises/practice e.g.:
`ExercismGenerator generateForExercise: 'darts'`
3. commit (each exercise own PR)

**Comment by glennj** [comment]
It'll take me a day or two to review this.

I look forward to contributing new exercises!

**Comment by Bajger** [comment]
@glennj Hi! Let me know, if I can merge this PR. Thanks in advance!
